src/wire-protocol.ts

```diff
@@ -1,11 +1,60 @@
+import { fql } from "./query-builder";
+import { TaggedTypeEncoded } from "./tagged-type";
+
+/**
+ * A piece of an interpolated query that represents an actual value. Arguments
+ * are passed to fauna using ValuFragments so that query string injection is not
+ * possible.
+ * @remarks A ValueFragment is created by this driver when a literal value or
+ * object is provided as an argument to the {@link fql} tagged template
+ * function.
+ *
+ * ValueFragments must always be encoded with tags, regardless of the "x-format"
+ * request header sent.
+ * @example
+ * ```typescript
+ *  const num = 17;
+ *  const query = fql`${num} + 3)`;
+ *  // produces
+ *  { fql: [{ value: { "@int": "17" } }, " + 3"] }
+ * ```
+ */
+export type ValueFragment = { value: TaggedTypeEncoded };
```

Comment:
Nit:

Thinking of readers here trying to build their own drivers - it will be easier to read if the "public" types appear first and the "private" types appear last:

- Please move these type definitions to the bottom of the file; or at least below the request / response models. (Alternatively, put them in another file). Do what you think works best.

Reply:
I dropped them down just above errors. I think I'd like to get errors moved into it's own module too, part of some reorganizing and dealing with properly exporting stuff to users.